Tests/SentryTests/Integrations/Performance/SentrySubClassFinderTests.swift

```diff
@@ -1,102 +1,74 @@
         }
     }
     
     private var fixture: Fixture!
-
+    
     override func setUp() {
         super.setUp()
         fixture = Fixture()
     }
-
-    func testActOnSubclassesOfParent_ReturnsChildren() {
-        testActOnSubclassesOf(Parent.self, expected: [Child1.self, Child2.self, GrandChild1.self, GrandChild2.self])
-    }
-    
-    func testActOnSubclassesOfChild1_ReturnsChildren() {
-        testActOnSubclassesOf(Child1.self, expected: [GrandChild2.self, GrandChild1.self])
-    }
-
-    func testActOnSubclassesOfChild2_ReturnsChildren() {
-        testActOnSubclassesOf(Child2.self, expected: [])
-    }
-    
-    func testActOnSubclassesOfChild2_WhenNewClassesRegistered_ReturnNoChildren() {
-        fixture.runtimeWrapper.beforeGetClassList = {
-            SentryClassRegistrator.registerClass(SentryId().sentryIdString)
-        }
-        testActOnSubclassesOf(Child1.self, expected: [])
-        XCTAssertEqual(0, fixture.runtimeWrapper.iterateClassesInvocations)
-    }
     
-    func testActOnSubclassesOfChild2_WhenNewClassOnlyOnceRegistered_ReturnsChildren() {
-        var invocations = 0
-        fixture.runtimeWrapper.beforeGetClassList = {
-            if invocations == 1 {
-                SentryClassRegistrator.registerClass(SentryId().sentryIdString)
-            }
-            invocations += 1
-        }
-        testActOnSubclassesOf(Child1.self, expected: [GrandChild2.self, GrandChild1.self])
-    }
-    
-    func testActOnSubclasses_SecondClassListReturns0_NoChildrenFound() {
-        var invocations = -1
-        fixture.runtimeWrapper.numClasses = { numClasses in
-            invocations += 1
-            return invocations > 0 ? 0 : numClasses
-        }
-        
-        testActOnSubclassesOf(Child1.self, expected: [])
-    }
-    
-    func testActOnSubclasses_SecondClassListReturnsOneLess_ReturnsChildren() {
-        var invocations = -1
-        fixture.runtimeWrapper.numClasses = { numClasses in
-            invocations += 1
-            return invocations > 0 ? numClasses - 1 : numClasses
-        }
-        
-        testActOnSubclassesOf(Child1.self, expected: [GrandChild2.self, GrandChild1.self])
+    func testActOnSubclassesOfViewController() {
+        testActOnSubclassesOfViewController(expected: [FirstViewController.self, SecondViewController.self])
     }
     
-    func testActOnSubclasses_ClassListKeepsReturnsOneLess_ReturnsNoChildren() {
-        var invocations = -1
-        fixture.runtimeWrapper.numClasses = { numClasses in
-            invocations += 1
-            return numClasses - Int32(invocations)
-        }
-        
-        testActOnSubclassesOf(Child1.self, expected: [])
-        XCTAssertEqual(0, fixture.runtimeWrapper.iterateClassesInvocations)
+    func testActOnSubclassesOfViewController_NoViewController() {
+        fixture.runtimeWrapper.classesNames = { _ in [] }
```

Comment:
`h`: You asked me how to write tests for the current implementation. I think you almost have it. The testActOnSubclassesOfViewController already makes sure that everything is called on the proper threads and also validates if the block is called on the right classes. I think we just need to pass in some class names to the runtime wrapper here and then adapt the expected below. Do this for all kinds of edge cases and we are good.